src/main/bootstrap/try_peer.rs

```diff
@@ -80,9 +89,17 @@ impl<UID: Uid> TryPeer<UID> {
             Ok(Some(Message::BootstrapGranted(peer_uid))) => {
                 let _ = core.remove_state(self.token);
                 let token = self.token;
-                let socket = mem::replace(&mut self.socket, Default::default());
-                let data = (socket, self.peer, peer_uid);
-                (*self.finish)(core, poll, token, Ok(data));
+
+                let mut socket = mem::replace(&mut self.socket, Default::default());
+                if socket
```

Comment:
It might be better to use match in these cases to log errors (in debug!()) to help debuggability in case something goes wrong.